### PR TITLE
resolved-ts: speed up advancing when stores get partitioned (#15567)

### PR DESCRIPTION
**Pull request by zyguan**
### What is changed and how it works?

Issue Number: ref #15567

What's Changed: pick #15567 to the standalone branch

```commit-message
```

### Related changes

- PR to update `pingcap/docs`/`pingcap/docs-cn`:
- Need to cherry-pick to the release branch

### Check List 

Tests 

- Unit test
- Integration test
- Manual test (add detailed scripts or steps below)
- No code

Side effects

- Performance regression
    - Consumes more CPU
    - Consumes more MEM
- Breaking backward compatibility

### Release note 

```release-note
None
```

### CODE REVIEW

**Review by you06 — APPROVED** [review]

File: components/resolved_ts/src/advance.rs [hunk 1]
```diff
@@ -435,10 +425,16 @@ impl LeadershipResolver {
             match res {
                 Ok((to_store, resp)) => {
                     for region_id in resp.regions {
-                        resp_map
-                            .entry(region_id)
-                            .or_insert_with(|| Vec::with_capacity(store_count))
-                            .push(to_store);
+                        if let Some(prog) = progresses.get_mut(&region_id) {
+                            if prog.resolved {
```
Comment (cfzjywxk) [inline components/resolved_ts/src/advance.rs:429]: Should the quorum also be checked when the `res` is `Err(...)` at https://github.com/tikv/tikv/pull/16036/files#diff-f86dc21a7845465caa70ed3ebe51652da2bb799b200680dd5f210701bb0c96d9R441?
For now it seems the `regiond` ids for each store RPC is not recorded yet.

Comment (zyguan) [reply components/resolved_ts/src/advance.rs:429]: This PR is just the cherry-pick of #15567. It's only add a short circuit when all regions are resolved. I think we can check `res` is `Err` and keep track a list of regions that we are sure they cannot be resolved, then break on `valid_regions.len() + invalid_regions.len() >= progresses.len()`. How about to do this optimization in next PR?

**Review by cfzjywxk — APPROVED** [review]

**Comment by cfzjywxk** [comment]
@zyguan The DCO needs to be fixed.

**Pull request merged**